File: PubeoAPI/Model/Vehicule.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PubeoAPI.model
{
    public class Vehicule
    {

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public String Marque { get; set; }
        public String Modele { get; set; }
        public Guid ParticulierId { get; set; }

        [ForeignKey("ParticulierId")]
        public virtual Particulier Particulier { get; set; }

        [Timestamp]
        public byte[] RowVersion { get; set; }
    }
}

File: PubeoApp/app/src/main/java/com/example/pubeo/model/Vehicule.java
package com.example.pubeo.model;

public class Vehicule {

    private String id;
    private String marque;
    private String modele;
    private String particulierId;
    private Particular particulier;

    public Vehicule(String id, String marque, String modele, String particulierId, Particular particulier) {
        this.id = id;
        this.marque = marque;
        this.modele = modele;
        this.particulierId = particulierId;
        this.particulier = particulier;
    }

    public String getId() {
        return id;
    }

    public String getMarque() {
        return marque;
    }

    public String getModele() {
        return modele;
    }

    public String getParticulierId() {
        return particulierId;
    }

    public Particular getParticulier() {
        return particulier;
    }
}
